swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items_matching_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                items_matching_categories.append(item)
+        return items_matching_categories
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.append(my_item)
+            friend.inventory.remove(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if self.inventory and friend.inventory:
+            if self.swap_items(friend, self.inventory[0], friend.inventory[0]):
+                return True
+        else:
+            return False
```

Comment:
We definitely need to make sure the inventories aren't empty, otherwise the `[0]` lookup could crash. Again, we can invert the check into a guard clause. We can also observe that the return value of `swap_first_item` is expected to behave the same as `swap_items`, so we can return the result from `swap_items` directly.

```python
    def swap_first_item(self, friend):
        if not self.inventory or not friend.inventory:
            return False

        return self.swap_items(friend, self.inventory[0], friend.inventory[0]):
```

Notice that due to the remove operations in `swap_items`, it is a linear operation, regardless of where in the list each item is. We could achieve constant time here either by swapping the items directly, or by doing an `index` then swap approach in `swap_items`.